tests/test_pytorch_sparse_dataloader_api.py

```diff
@@ -0,0 +1,249 @@
+"""Tests for TileDB integration with Pytorch Data API."""
+
+import torch
+import tiledb
+import numpy as np
+import unittest
+import tempfile
+
+import torch.nn as nn
+import torch.optim as optim
+
+from tiledb.ml.data_apis.pytorch_sparse import PyTorchTileDBSparseDataset
+
+# Test parameters
+NUM_OF_CLASSES = 5
+BATCH_SIZE = 20
+ROWS = 1000
+
+# We test for 2d
+INPUT_SHAPES = [
+    (10,),
+]
+# We test for single and multiple workers
+# TODO#1: Multiple workers require tiledb.SparseArray to be pickled hence serializable as well
+NUM_OF_WORKERS = [0]
+
+
+def create_sparse_array_one_hot_2d(rows: int, cols: tuple) -> np.ndarray:
+    seed = np.random.randint(low=0, high=cols[0], size=(rows,))
+    seed[-1] = cols[0] - 1
+    b = np.zeros((seed.size, seed.max() + 1))
+    b[np.arange(seed.size), seed] = 1
+    return b
+
+
+def get_schema(data: np.array, batch_size: int, sparse: bool) -> tiledb.ArraySchema:
+    dims = [
+        tiledb.Dim(
+            name="dim_" + str(dim),
+            domain=(0, data.shape[dim] - 1),
+            tile=data.shape[dim] if dim > 0 else batch_size,
+            dtype=np.int32,
+        )
+        for dim in range(data.ndim)
+    ]
+
+    # TileDB schema
+    schema = tiledb.ArraySchema(
+        domain=tiledb.Domain(*dims),
+        sparse=sparse,
+        attrs=[tiledb.Attr(name="features", dtype=np.float32)],
+    )
+
+    return schema
+
+
+def ingest_in_tiledb(data: np.array, batch_size: int, uri: str):
+    schema = get_schema(data, batch_size, False)
+
+    # Create array
+    tiledb.Array.create(uri, schema)
+
+    # Ingest
+    with tiledb.open(uri, "w") as tiledb_array:
+        tiledb_array[:] = {"features": data}
+
+
+def ingest_in_tiledb_sparse(data: np.array, batch_size: int, uri: str):
+    schema = get_schema(data, batch_size, True)
+
+    # Create the (empty) array on disk.
+    tiledb.Array.create(uri, schema)
+
+    # Ingest
+    with tiledb.open(uri, "w") as tiledb_array:
+        idx = np.nonzero(data)
+        I, J = idx[0], idx[1]
+        tiledb_array[I, J] = {"features": data[idx]}
```

Comment:
1. Don't need to unpack `idx`, just use it as is (`tiledb_array[idx]`)
2. This function can be merged with `ingest_in_tiledb` like this:
```py
def ingest_in_tiledb(data: np.array, batch_size: int, uri: str, sparse: bool):
     ...
    with tiledb.open(uri, "w") as tiledb_array:
        idx =  np.nonzero(data) if sparse else slice(None)
        tiledb_array[idx] = {"features": data[idx]}
```

Reply:
Unfortunately current tiledb==0.8.8 throws an error when I try that:

```
tiledb_array[idx] = {"features": data[idx]}
  File "tiledb/libtiledb.pyx", line 5264, in tiledb.libtiledb.SparseArrayImpl.__setitem__
  File "tiledb/libtiledb.pyx", line 5162, in tiledb.libtiledb._setitem_impl_sparse
  File "tiledb/libtiledb.pyx", line 5138, in genexpr
  File "tiledb/libtiledb.pyx", line 5138, in genexpr
TypeError: int() argument must be a string, a bytes-like object or a number, not 'slice'

```

Do we support it in next versions?